src/main/java/io/projectriff/invoker/server/GrpcServerAdapter.java

```diff
@@ -67,11 +69,20 @@ public GrpcServerAdapter(FunctionCatalog functionCatalog, String functionName) {
                             .map(this::toSpringMessage)
                             .transform(invoker(userFn))
                             .map(this::fromSpringMessage)
-                            .onErrorMap(e -> Status.UNKNOWN.withDescription(e.getMessage()).withCause(e).asException())
+                            .onErrorMap(this::handleConversionExceptions)
                             .doOnError(Throwable::printStackTrace);
                 });
     }
 
+    private StatusException handleConversionExceptions(Throwable e) {
+        if (e instanceof MessageConversionException && e.getMessage().equals(BeanFactoryAwareFunctionRegistry.COULD_NOT_CONVERT_INPUT)) {
+            return Status.INVALID_ARGUMENT.withDescription("Invoker: Unsupported Media Type: " + e.getMessage()).withCause(e).asException();
```

Comment:
as for the Node invoker, it would probably be good to extract this prefix (`"Invoker: Unsupported Media Type"`) to a constant.